### PR TITLE
Preprocess interface + implementations

### PR DESCRIPTION
**Pull request by ASalvail**
First draft of `Preprocess` class and a couple implementations.

Any thoughts @MarcCote ?

### CODE REVIEW

**Comment by ASalvail** [comment]
This should do the trick!

**Comment by MarcCote** [comment]
This seems perfect but... could you add two small unit tests? One for `NormalizeFeature` and `LagExample`. That would _really_ help me sleep at night :P.

**Comment by ASalvail** [comment]
Addressed @MarcCote comment. I removed the `LagExample` preprocessor because it really wasn't working as intended. (I actually don't know what it was doing...)

File: smartlearner/preprocessors.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import numpy as np
+
+from .interfaces.preprocess import Preprocess
+
+
+class NormalizeFeature(Preprocess):
```
Comment (MarcCote) [inline smartlearner/preprocessors.py:6]: We should say what type of normalization it is. According to https://en.wikipedia.org/wiki/Normalization_(statistics) it is "Normalizing errors when population parameters are known. Works well for populations that are normally distributed".

**Pull request merged**